pkg/services/app/add.go

```diff
@@ -633,6 +694,59 @@ func generateResourceName(url string) string {
 	return hashNameIfTooLong(strings.ReplaceAll(utils.UrlToRepoName(url), "_", "-"))
 }
 
+func (a *App) createKeepFilePullRequest(info *AppResourceInfo, cloneNeeded bool) error {
```

Comment:
This was a bit alarming. Can't we just add the `.keep` file to the list of files we already commit?

Reply:
Not really. The .keep file is in the app repo while the other files are in the config repo. We don't need to do this at all in the default case where the automation is stored in the _app_ repo; only in the `--app-config-url=NONE` and `--app-config-url=<url>` case.